source/DasBlog.Web.UI/Startup.cs

```diff
@@ -120,6 +124,14 @@ public void ConfigureServices(IServiceCollection services)
 				.AddTransient<IPrincipal>(provider => provider.GetService<IHttpContextAccessor>().HttpContext.User)
 				.AddTransient<ISiteRepairer, SiteRepairer>()
 				;
+			services
+				.AddScoped<IRichEditBuilder>(provider
```

Comment:
I think maybe using a swtich statement would be better. that way if someone wants to add anotehr rich editor, it won't be difficult to read.